mantidproject/mantid
# Add fit output status to Indirect Data Analysis fitting tabs

**Description of work.**
This PR adds the fit output information (algorithm status and chi squared) to each fitting tab within the Indirect Data Analysis Interface. In order to break the existing API, these two outputs are optional, which is controlled through an additional flag to the PlotPeaksByLogValue Algorithm (and the QENSFit algorithms).

**Report to:** ISIS/Spencer

**To test:**
Data to test each tab of the Indirect Data Analysis GUI can be found in #26631.

1. Open the Indirect Data Analysis GUI.
2. Go to the ``Conv fit`` tab
3. Load in data
4. Add the ``One Lorentzian`` function
5. Do a fit - A fit success and chi squared value should be shown.
6. You can cycle through the chi squared values and fit status using the ``Plot Spectrum`` spinbox 
7. Change the ``Max Iterations`` to 1 and run the fit - A fit failed message should be shown

Lastly, check that the functionally works on the other fitting tabs.


Fixes #28964

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
